pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go

```diff
@@ -198,3 +203,26 @@ type MetricsStatus struct {
 	// TargetValue indicates the threshold value for this metrics in auto-scaling
 	ThresholdValue string `json:"thresholdValue"`
 }
+
+// +k8s:openapi-gen=true
+// ExternalEndpoint describe the external service enpoint
+// who provide the tikv/tidb auto-scaling recommended replicas ability
+type ExternalEndpoint struct {
+	// Host indicates the external service's host
+	Host string `json:"host"`
+	// Port indicates the external service's post
+	Port int32 `json:"port"`
+	// Path indicates the external service's path
+	Path string `json:"path"`
+	// TlsSecretRef indicates the Secret which stored the tls configuration. If enabled, operator would use https
+	// to communicate to the external service
+	// +optional
+	TlsSecretRef *SecretRef `json:"tlsSecret,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// SecretRef indicates to secret ref
+type SecretRef struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
```

Comment:
If not set, we can take the namespace of tidb-controller-manager.

Reply:
I think it is not necessary and didn't make sense.